renaissance-harness/src/main/java/org/renaissance/harness/ExecutionDriver.java

```diff
@@ -1,103 +1,112 @@
 package org.renaissance.harness;
 
 import org.renaissance.Benchmark;
-import org.renaissance.BenchmarkContext;
-import org.renaissance.BenchmarkParameter;
 import org.renaissance.BenchmarkResult;
 import org.renaissance.BenchmarkResult.ValidationException;
 import org.renaissance.Plugin.ExecutionPolicy;
-import org.renaissance.core.BenchmarkInfo;
+import org.renaissance.core.BenchmarkDescriptor;
+import org.renaissance.core.BenchmarkSuite;
+import org.renaissance.core.BenchmarkSuite.SuiteBenchmarkContext;
 
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Path;
 import java.util.Locale;
 
-import static java.nio.file.Files.createDirectories;
-
 /**
  * Benchmark execution driver. Captures the sequence of actions performed
  * during benchmark execution (i.e., calling the sequencing methods on a
  * benchmark and issuing notifications to event listeners) and maintains
  * execution context.
  */
-final class ExecutionDriver implements BenchmarkContext {
+final class ExecutionDriver {
+
+  /** Context for the executed benchmark. */
+  private final SuiteBenchmarkContext benchmarkContext;
+
+  /** Name of the benchmark. */
+  private final String benchmarkName;
 
-  /** Information about the benchmark to execute. */
-  private final BenchmarkInfo benchInfo;
+  /** The benchmark to execute. */
+  private final Benchmark benchmark;
 
-  /** Name of the benchmark configuration to use. */
-  private final String configName;
+  /** The dispatcher for execution-related events. */
+  private final EventDispatcher eventDispatcher;
 
-  /** Harness scratch root directory. */
-  private final Path scratchRoot;
+  /** The policy to control operation execution. */
+  private final ExecutionPolicy executionPolicy;
+
+  /** Preformatted message to print before executing the benchmark. */
+  private final String beforeEachMessageFormat;
+
+  /** Preformatted message to print after executing the benchmark. */
+  private final String afterEachMessageFormat;
 
   /** The value of {@link System#nanoTime()} at VM start. */
   private final long vmStartNanos;
 
-  /** Benchmark scratch directory. */
-  private Path scratchDir;
 
-  public ExecutionDriver(
-    final BenchmarkInfo benchInfo, final String configName,
-    final Path scratchRoot, final long vmStartNanos
+  private ExecutionDriver(
+    final SuiteBenchmarkContext context,
+    final Benchmark benchmark,
+    final EventDispatcher dispatcher,
+    final ExecutionPolicy policy,
+    final long vmStartNanos
   ) {
-    this.benchInfo = benchInfo;
-    this.configName = configName;
-    this.scratchRoot = scratchRoot;
+    this.benchmarkContext = context;
+    this.benchmark = benchmark;
+    this.eventDispatcher = dispatcher;
+    this.executionPolicy = policy;
     this.vmStartNanos = vmStartNanos;
+
+    this.benchmarkName = context.benchmarkName();
+    this.beforeEachMessageFormat = String.format(
+      "====== %s (%s) [%s], iteration %%d started ======\n",
```

Comment:
Oh, I got confused by the escaped percent, but you're actually just preparing a string to format with a string format :)

Reply:
I wanted to preformat the stable things to avoid passing it down the print functions over and over.